Tests/NIOTransportServicesTests/NIOTSChannelOptionsTests.swift

```diff
@@ -0,0 +1,115 @@
+    }
+    
+    func testCurrentPath() throws {
+        let listener = try NIOTSListenerBootstrap(group: self.group)
+            .bind(host: "localhost", port: 0).wait()
+        defer {
+            XCTAssertNoThrow(try listener.close().wait())
+        }
+
+        let connection = try NIOTSConnectionBootstrap(group: self.group)
+            .connect(to: listener.localAddress!)
+            .wait()
+        defer {
+            XCTAssertNoThrow(try connection.close().wait())
+        }
+        
+        let currentPath = try connection.getOption(NIOTSChannelOptions.currentPath).wait()
+        XCTAssertEqual(currentPath.status, NWPath.Status.satisfied)
+    }
+    
+    @available(OSX 10.15, iOS 12.0, tvOS 12.0, watchOS 6.0, *)
+    func testMetadata() throws {
+        let listener = try NIOTSListenerBootstrap(group: self.group)
+            .bind(host: "localhost", port: 0).wait()
+        defer {
+            XCTAssertNoThrow(try listener.close().wait())
+        }
+
+        let connection = try NIOTSConnectionBootstrap(group: self.group)
+            .connect(to: listener.localAddress!)
+            .wait()
+        defer {
+            XCTAssertNoThrow(try connection.close().wait())
+        }
+        
+        let metadata = try connection.getOption(NIOTSChannelOptions.metadata(NWProtocolTCP.definition)).wait() as! NWProtocolTCP.Metadata
+        XCTAssertEqual(metadata.availableReceiveBuffer, 0)
+    }
+
+    @available(OSX 10.15, iOS 12.0, tvOS 12.0, watchOS 6.0, *)
+    func testEstablishmentReport() throws {
+        let listener = try NIOTSListenerBootstrap(group: self.group)
+            .bind(host: "localhost", port: 0).wait()
+        defer {
+            XCTAssertNoThrow(try listener.close().wait())
+        }
+
+        let connection = try NIOTSConnectionBootstrap(group: self.group)
+            .connect(to: listener.localAddress!)
+            .wait()
+        defer {
+            XCTAssertNoThrow(try connection.close().wait())
+        }
+        
+        let reportFuture = try connection.getOption(NIOTSChannelOptions.establishmentReport).wait()
+        let establishmentReport = try reportFuture.wait()
+        
+        XCTAssertEqual(establishmentReport!.resolutions.count, 0)
+    }
+    
+    @available(OSX 10.15, iOS 12.0, tvOS 12.0, watchOS 6.0, *)
+    func testDataTransferReport() throws {
+        let listener = try NIOTSListenerBootstrap(group: self.group)
+            .bind(host: "localhost", port: 0).wait()
+        defer {
+            XCTAssertNoThrow(try listener.close().wait())
+        }
+
+        let connection = try NIOTSConnectionBootstrap(group: self.group)
+            .connect(to: listener.localAddress!)
+            .wait()
+        defer {
+            XCTAssertNoThrow(try connection.close().wait())
+        }
+        
+        let pendingReport = try connection.getOption(NIOTSChannelOptions.dataTransferReport).wait()
+        
+        pendingReport.collect(queue: .main) { report in
```

Comment:
Let's try collecting this on a new one-off serial queue and using `DispatchGroup` to wait for the result. Right now we don't wait for this to complete, but doing that should work well.